cmd/nginx-ingress/flags.go

```diff
@@ -0,0 +1,400 @@
+
+	glog.Infof("Starting NGINX Ingress Controller %v PlusFlag=%v", versionInfo, *nginxPlus)
+	glog.Info(binaryInfo)
+
+	validationChecks()
+
+	if *enableTLSPassthrough && !*enableCustomResources {
+		glog.Fatal("enable-tls-passthrough flag requires -enable-custom-resources")
+	}
+
+	if *enablePreviewPolicies {
+		glog.Warning("enable-preview-policies is universally deprecated. To enable OIDC Policies please use -enable-oidc instead.")
+	}
+	*enableOIDC = *enablePreviewPolicies || *enableOIDC
+
+	if *appProtect && !*nginxPlus {
+		glog.Fatal("NGINX App Protect support is for NGINX Plus only")
+	}
+
+	if *appProtectLogLevel != appProtectLogLevelDefault && !*appProtect && !*nginxPlus {
+		glog.Fatal("app-protect-log-level support is for NGINX Plus only and App Protect is enable")
+	}
+
+	if *appProtectDos && !*nginxPlus {
+		glog.Fatal("NGINX App Protect Dos support is for NGINX Plus only")
+	}
+
+	if *appProtectDosDebug && !*appProtectDos && !*nginxPlus {
+		glog.Fatal("NGINX App Protect Dos debug support is for NGINX Plus only and App Protect Dos is enable")
+	}
+
+	if *appProtectDosMaxDaemons != 0 && !*appProtectDos && !*nginxPlus {
+		glog.Fatal("NGINX App Protect Dos max daemons support is for NGINX Plus only and App Protect Dos is enable")
+	}
+
+	if *appProtectDosMaxWorkers != 0 && !*appProtectDos && !*nginxPlus {
+		glog.Fatal("NGINX App Protect Dos max workers support is for NGINX Plus and App Protect Dos is enable")
+	}
+
+	if *appProtectDosMemory != 0 && !*appProtectDos && !*nginxPlus {
+		glog.Fatal("NGINX App Protect Dos memory support is for NGINX Plus and App Protect Dos is enable")
+	}
+
+	if *spireAgentAddress != "" && !*nginxPlus {
+		glog.Fatal("spire-agent-address support is for NGINX Plus only")
+	}
+
+	if *enableInternalRoutes && *spireAgentAddress == "" {
+		glog.Fatal("enable-internal-routes flag requires spire-agent-address")
+	}
+
+	if *enableLatencyMetrics && !*enablePrometheusMetrics {
+		glog.Warning("enable-latency-metrics flag requires enable-prometheus-metrics, latency metrics will not be collected")
+		*enableLatencyMetrics = false
+	}
+
+	if *enableCertManager && !*enableCustomResources {
+		glog.Fatal("enable-cert-manager flag requires -enable-custom-resources")
+	}
+
+	if *ingressLink != "" && *externalService != "" {
+		glog.Fatal("ingresslink and external-service cannot both be set")
+	}
+}
+
+func initialChecks() {
+	err := flag.Lookup("logtostderr").Value.Set("true")
+	if err != nil {
+		glog.Fatalf("Error setting logtostderr to true: %v", err)
+	}
+
+	if startupCheckFn != nil {
+		err := startupCheckFn()
+		if err != nil {
+			glog.Fatalf("Failed startup check: %v", err)
+		}
+	}
+
+	unparsed := flag.Args()
+	if unparsed != nil {
+		glog.Warningf("Ignoring unhandled arguments: %v", unparsed)
+	}
+}
+
+// printVersionInfo prints the the version and binary info before exiting if the flag is set
+func printVersionInfo(versionInfo string, binaryInfo string) {
+	if *versionFlag {
```

Comment:
I would have this if outside of the `func`, but also I never liked the duplication for printing the info, I think we could probably always print the info and just exit on `--version` in the future.